path_finding.asciidoc

```diff
@@ -199,19 +210,65 @@ But wait: Selena does know *some* channel balances! She knows the balances of th
 .Channel graph with known and unknown balances
 image::images/channel_graph_2.png[]
 
+While the "?" symbol seems ominous, a lack of certainty is not the same as complete ignorance. We can _quantify_ the uncertainty and _reduce_ it by updating the graph with the successful/failed HTLCs we attempt.
+
+Uncertainty can be quantified, because we know the maximum and minimum possible liquidity and can calculate probabilities for smaller (more precise) ranges.
+
+Once we attempt to send an HTLC we can learn more about channel balances: if we succeed, then the balance was *at least* sufficient to transport the specific amount. Meanwhile if we get a "temporary channel failure" error, the most likely reason is a lack of liquidity for the specific amount.
+
+[TIP]
+====
+You may be thinking "What's the point of learning from a successful HTLC?" After all, if it succeeded we're "done". But consider that we may be sending one part of a multi-part payment. We also may be sending other single-part payments within a short time. Anything we learn about liquidity is useful for the next attempt!
```

Comment:
Even during a single part payment in the trial and error loop it is useful as an onion might have failed close to the receiving node. one can now start path finding from all upstream nodes for the next attempt. 

of course somewhere we should mention that the knowledge is just temporary